sale_coupon_order_line_link/migrations/16.0.1.0.0/post-migration.py

```diff
@@ -0,0 +1,15 @@
+# Copyright 2023 Tecnativa - Pilar Vargas
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from openupgradelib import openupgrade
+
+
+def _fill_reward_id_field(env):
+    for line in env["sale.order.line"].filtered("loyalty_program_id"):
+        line.reward_id = env["reward.program"].search(
+            [["program_id", "=", line.loyalty_program_id.id]]
+        )
```

Comment:
Better do this in pre and with sql for performance and to avoid to trigger ORM operations

Reply:
I doubted whether such a field will already exist in pre

Reply:
It probably won't. You should try to pre-create the column: 

```sql
        ALTER TABLE sale_order_line
        ADD COLUMN IF NOT EXISTS reward_id INTEGER;
```